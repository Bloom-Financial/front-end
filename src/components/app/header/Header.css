header {
  position: fixed;
  top: 0;
  height: 70px;
  background-color: rgb(197, 178, 178);
  width: 100%;
  display: flex;
  align-items: flex-end;
  justify-content: center;
  box-shadow: 1px 1px 1px 1px rgba(116, 110, 110, 0.199);
}
.BioLink {
  font-family: 'Fjalla One', sans-serif;
  font-size: 4rem;
  position: absolute;
  right: 10vw;
}
