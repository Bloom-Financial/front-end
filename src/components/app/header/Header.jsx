import React, { setState } from 'react';
import PropTypes from 'prop-types';

const Header = ({ username, onSubmit }) => {
  // const [username, setUsername ] = setState('');
  // const [profile, setProfile ] = setState('');

  return (
    <>
      <form>
        <input id="username" 
          type="text" 
          value={username}>username: {username} </input>
        <button onSubmit={onSubmit}>Update Profile</button>   
      </form>
      <button onSubmit={onSubmit}>login out</button>
    </>
  );
};

Header.propTypes = {
  username: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Header;

