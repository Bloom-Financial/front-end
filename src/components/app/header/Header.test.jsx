import React from 'react';
import { render } from '@testing-library/react';
import Header from './Header';


describe('Test the header component', () => {
  it('tests the header component', () => {
    const { asFragment } = render(<Header title="hey"/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
