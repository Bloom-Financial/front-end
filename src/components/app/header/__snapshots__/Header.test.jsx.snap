// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the header component tests the header component 1`] = `
<DocumentFragment>
  <form>
    <label
      for="username"
    >
      Enter username
    </label>
    <input
      id="username"
      type="text"
    />
    <button>
      Update Profile
    </button>
  </form>
  <button>
    log out
  </button>
</DocumentFragment>
`;
